src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java21/InputStringTemplateNested.java

```diff
@@ -8,7 +8,13 @@ public class InputStringTemplateNested {
     final String s =
             STR."x\{ sp(() -> {
                 return STR."x\{ sp(() -> {
-                    return STR."x\{ x }x" + "{" + "}}}";
+                    return STR."x\{
+                            sp(() -> {
+        return sp(() -> {
+            return sp(() -> {
+                return sp(() -> {
+                    return sp(() -> "");});});});})
+                            }x" + "{" + "}}}";
```

Comment:
Previous test case only went to a nesting level of 1, we needed to go higher to kill a pitest mutation (and be more thorough in our testing in general).